### ISSUE TITLE
Unassign Deactivated Volunteers from Supervisors Automatically

### ISSUE THREAD

**Issue by CovenantHuman** [rubyforgood/casa#3333]
**What type(s) of user does this feature affect?**
 - supervisors
 
**Description**
Currently when a volunteer is deactivated they stay assigned to their supervisor until manually unassigned. We would like to change this so that when a volunteer is deactivated they are automatically unassigned from their supervisor. 

**Screenshots of current behavior, if any**  
Here is an example deactivated volunteer:
![Deactivated Volunteer](https://user-images.githubusercontent.com/7007647/161337104-d9285807-6bd5-4dfb-a9b6-945605afa70c.png)
Here they are shown still assigned to their supervisor:
![Deactivated Volunteer Still Assigned](https://user-images.githubusercontent.com/7007647/161337168-99e77735-0e6b-458c-8fe6-f660d59862d0.png)

_QA Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

**Comment by MrJackStone** [comment]
Hey! We'd like to work on this feature.

**Comment by compwron** [comment]
@MrJackStone go for it! :)

**Comment by MrJackStone** [comment]
I have a question! After making the changes we deemed necessary one unrelated test broke (see image below). In attempting to debug it we tried to login as all_casa_admin (allcasaadmin@example.com, all_casa_admin1@example.com, admin1@example.com), but failed every time. We wanted to login as all_casa_admin to be able to visualize the webpage that failed the test in the browser. Is there any reason why we can't login as all_casa_admin?

Thanks in advance! :D

![image](https://user-images.githubusercontent.com/70590511/164761224-1983d8e6-0703-481c-a8e1-2c201766d3ce.png)
